### ISSUE TITLE
Renaming 'AssemblyName' for a default Analyzer + Fix project causes a build failure

### ISSUE THREAD

**Issue by mavasani** [dotnet/roslyn#3516]
1. Create a new C# Analyzer + Fix project from the template (my project name was `Analyzer5` for the repro)
2. Build and verify it succeeds.
3. Open project property page for the Analyzer project and rename the `Assembly Name`
4. Build again

**Got:**
```
Severity	Code	Description	Project	File	Line	Build tool
Error		Cannot get assembly name for "bin\Debug\\Analyzer5.dll". Could not load file or assembly 'Analyzer5.dll' or one of its dependencies. The system cannot find the file specified.	Analyzer5	c:\users\mavasani\documents\visual studio 2015\Projects\Analyzer5\Analyzer5\Analyzer5\Analyzer5.csproj	122	
```
The analyzer nugget package is not generated in the output directory.

I think the issue is with the `AfterBuild` target in the project file. It should be referencing the `$(AssemblyName)` variable instead of having a hardcoded assembly name from the project creation time:

```
<Target Name="AfterBuild">
    <GetAssemblyIdentity AssemblyFiles="$(OutDir)\Analyzer5.dll">
      <Output TaskParameter="Assemblies" ItemName="Analyzer5AssemblyInfo" />
    </GetAssemblyIdentity>
    <Exec Command="&quot;$(SolutionDir)packages\NuGet.CommandLine.2.8.5\tools\NuGet.exe&quot; pack Diagnostic.nuspec -NoPackageAnalysis -Version %(Analyzer5AssemblyInfo.Version) -OutputDirectory ." WorkingDirectory="$(OutDir)" LogStandardErrorAsError="true" ConsoleToMSBuild="true">
      <Output TaskParameter="ConsoleOutput" PropertyName="OutputOfExec" />
    </Exec>
  </Target>
```

**Comment by jmarolf** [comment]
This was fixed with [1e078d15](https://github.com/dotnet/roslyn-internal/commit/1e078d1513d9db5ec7d7e87ec63e1aa458fe3a0e)

**Comment by jmarolf** [comment]
@mavasani could you verify that this was fixed?
